hypixel_api_lib/Items.py

```diff
@@ -0,0 +1,102 @@
+import requests
+
+ITEMS_API_URL = r"https://api.hypixel.net/v2/resources/skyblock/items"
+
+class SkyBlockItem:
+    """
+    Represents an item in SkyBlock.
+
+    Attributes:
+        id (str): The unique identifier for the item.
+        material (str): The Bukkit material enum value for the item.
+        name (str): The name of the item.
+        tier (str): The rarity tier of the item.
+        color (str, optional): The color metadata to be applied to the item.
+        skin (str, optional): The skin value for a skull-based item.
+    """
+    
+    def __init__(self, id, material, name, tier, color=None, skin=None):
+        self.id = id
+        self.material = material
+        self.name = name
+        self.tier = tier if tier is not None else 'UNKNOWN'
+        self.color = color
+        self.skin = skin
+
+    def __str__(self):
+        return f"{self.name} ({self.tier}): ID={self.id}, Material={self.material}"
+
+class Items:
+    """
+    Handles fetching and managing all the items from the API.
+    
+    Attributes:
+        api_endpoint (str): The endpoint URL to fetch the items data.
+        items (dict of str: SkyBlockItem): A dictionary of item IDs to SkyBlockItem objects.
+    """
+    
+    def __init__(self, api_endpoint=ITEMS_API_URL):
```

Comment:
nice touch

Reply:
@desobob11 You think its good to merge now then? Approve if you are happy with it and then we can complete this PR